viewer/packages/cad-styling/src/prioritized/BoxClusterer.ts

```diff
@@ -29,21 +29,37 @@ export class SmartMergeBoxes implements BoxClusterer {
     this.resultBoxes.pop();
   }
 
-  private shouldMergeBoxesAtIndices(i: number, j: number): boolean {
-    const BOX_MERGE_MIN_IOU = 0.15;
-    const overlap = intersectionOverUnion(this.resultBoxes[i], this.resultBoxes[j]);
-    return overlap >= BOX_MERGE_MIN_IOU;
+  private surfaceArea(box: Box3) {
+    const size = box.getSize(new Vector3());
+
+    return 2 * (size.x * size.y + size.y * size.z + size.z * size.x);
+  }
+
+  private shouldMergeBoxes(box0: Box3, box1: Box3): boolean {
+    const MAX_SURFACE_INCREASE_RATIO = 1.0;
+    const MAX_SURFACE_INCREASE_ADDITIVE_TERM = 8.0; // Heuristic number of square meters to allow merging of smaller boxes
+
+    const box0Copy = this.reusableBoundingBox.copy(box0);
```

Comment:
See https://github.com/cognitedata/reveal/blob/master/viewer/packages/cad-geometry-loaders/src/sector/culling/OrderSectorsByVisibilityCoverage.ts#L276 for how we've done this before - probably good to keep the pattern